tdms/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+project(tdms)
+
+# General setup ---------------------------------------------------------------
+cmake_minimum_required(VERSION 3.22)
+
+# Allow RPATH on mac
+set(CMAKE_MACOSX_RPATH 1)
+
+# Append the cmake/ directory to the search path
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
+
+# and the  include directories for the project
+include_directories(${CMAKE_SOURCE_DIR}/include)
+include_directories(${CMAKE_SOURCE_DIR}/matlabio/)
+
+
+# Command line options for CMake ----------------------------------------------
+# Type of derivatives to use
+set(DERIVATIVE_TYPE "PS" CACHE STRING "Type of derivatives to compute.
+Either Use pseudo spectral (PS) or finite differences (FD).
+Options: {FD, PS}")
+set_property(CACHE DERIVATIVE_TYPE PROPERTY STRINGS "FD" "PS")
+get_property(x CACHE DERIVATIVE_TYPE PROPERTY STRINGS)
+if (NOT DERIVATIVE_TYPE IN_LIST x)
+    message(FATAL_ERROR "Cannot set DERIVATIVE_TYPE=${DERIVATIVE_TYPE}. Options: ${x}")
+endif()
+
+if (${DERIVATIVE_TYPE} STREQUAL FD)
+    set(DFLAG "-DFDFLAG")
+endif()
+
+
+# Matlab ----------------------------------------------------------------------
+find_package(Matlab REQUIRED MAT_LIBRARY)
+include_directories(${Matlab_INCLUDE_DIRS})
+
+# Currently matlab only has x84 libraries on Mac (no arm)
+set(CMAKE_OSX_ARCHITECTURES "x86_64")
```

Comment:
Sounds like there is already a beta version native for Apple Silicon: https://blogs.mathworks.com/matlab/2022/05/05/exploring-the-matlab-beta-for-native-apple-silicon/.  Maybe you can try this one out?  This should simplify the build system a bit.

Reply:
I've spent a moment trying and [FindMatlab.cmake](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindMatlab.cmake#L1741) is hard coded for intel, and I'm disinclined to copy it into this project and modify or edit the upstream for something that is in beta 😕